esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -6118,7 +6104,289 @@ bool CWsDfuEx::onDFUFileAccess(IEspContext &context, IEspDFUFileAccessRequest &r
             userDesc.setown(createUserDescriptor());
             userDesc->set(userID.str(), context.queryPassword(), context.querySignature());
         }
-        getFileAccess(context, userDesc, accessType, req, resp);
+        getFileAccess(context, userDesc, accessType, requests.getName(), requests.getJobId(), requests, resp.updateAccessInfo());
+    }
+    catch (IException *e)
+    {
+        FORWARDEXCEPTION(context, e,  ECLWATCH_INTERNAL_ERROR);
+    }
+    return true;
+}
+
+const char *CWsDfuEx::createWsDFUCreateGroupName(StringArray &locations, StringBuffer &groupName)
+{
+    if (locations.length() < 1)
+        return nullptr;
+
+    StringBuffer buf;
+    ForEachItemIn(i, locations)
+        buf.append(locations.item(i)).append(";");
+    groupName.set("wsdfucreate").append(hashc((unsigned char*) buf.str(), buf.length(), 0));
+    return groupName.str();
+}
+
+IGroup *CWsDfuEx::queryDFUFileIGroup(StringBuffer &groupName, ClusterType clusterType, StringArray &locations)
+{
+    GroupType groupType;
+    StringBuffer basedir;
+    Owned<IGroup> group = queryNamedGroupStore().lookup(groupName.str(), basedir, groupType);
+    if (group)
+    {
+        if (groupType == grp_roxie)
+            throw MakeStringException(ECLWATCH_INVALID_INPUT, "IGroup for roxie not allowed: %s.", groupName.str());
```

Comment:
given this service is the one that's creating these with unique names, I don't think this check is relevant and only confuses.

Reply:
Ok, will remove